core/deployment/src/test/java/io/quarkus/deployment/CapabilityTest.java

```diff
@@ -0,0 +1,43 @@
+package io.quarkus.deployment;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class CapabilityTest {
+    private static Stream<Arguments> capabilityFields() {
+        Field[] declaredFields = Capability.class.getDeclaredFields();
+        return Stream.of(declaredFields)
+                .filter(field -> Modifier.isStatic(field.getModifiers()))
+                .filter(field -> field.getType().equals(String.class))
+                .filter(field -> !getString(field).equals(Capability.QUARKUS_PREFIX))
+                .map(field -> Arguments.of(field));
+    }
+
+    @ParameterizedTest
+    @MethodSource("capabilityFields")
+    void validateCapabilityString(Field field) {
+        if (Modifier.isStatic(field.getModifiers()) && field.getType().equals(String.class)) {
```

Comment:
This `if` is not needed, since you're already filtering it out above

Reply:
Actually, don't pass the `Field` here, just the String extracted from the arguments above

Reply:
You need to change the capabilityFields() method above to return Strings instead of fields

Reply:
ah sorry, I did that, but I forgott to stage it 🤦‍♂️